core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -391,13 +390,6 @@ private Optional<OrderingScheme> decorrelateOrderingScheme(OrderingScheme orderi
             return Optional.of(new OrderingScheme(nonConstantOrderBy.build(), nonConstantOrderings.buildOrThrow()));
         }
 
-        @Override
-        public Optional<DecorrelationResult> visitEnforceSingleRow(EnforceSingleRowNode node, Void context)
-        {
-            Optional<DecorrelationResult> childDecorrelationResultOptional = node.getSource().accept(this, null);
-            return childDecorrelationResultOptional.filter(result -> result.atMostSingleRow);
```

Comment:
from cmt msg

> EnforceSingleRowNode guarantees that exactly single row
is produced. Therefore, PlanNodeDecorrelator was incorrectly
removing EnforceSingleRowNode even if decorrelated subplan
could produce 0 rows.

sounds like we could improve the logic (instead of removing it),
if we knew the plan produces exactly single row (eg global aggregation)